service/src/main/java/greencity/service/locations/LocationApiService.java

```diff
@@ -0,0 +1,355 @@
+     * @param level    The hierarchical level of the locations to fetch (e.g., city,
+     *                 region).
+     * @return A List of LocationDto objects representing all locations at the
+     *         provided level.
+     */
+    private List<LocationDto> getLocationDataByLevel(int pageSize, int level) {
+        UriComponentsBuilder builder = builderUrl(pageSize).queryParam(LEVEL, level);
+        return getResultFromUrl(builder.build().encode().toUri());
+    }
+
+    /**
+     * Fetches location data by the provided name.
+     *
+     * @param pageSize The maximum number of results to return.
+     * @param level    The hierarchical level of the location to fetch (e.g., city,
+     *                 region).
+     * @param name     The name of the location to fetch.
+     * @return A List of LocationDto objects representing the locations matching the
+     *         provided name.
+     * @throws NotFoundException if no location matching the provided name and level
+     *                           can be found.
+     */
+    private List<LocationDto> getLocationDataByName(int pageSize, int level, String name) {
+        if (name == null) {
+            throw new IllegalArgumentException("The name parameter cannot be null");
+        }
+        UriComponentsBuilder builder = builderUrl(pageSize)
+            .queryParam(NAME, name)
+            .queryParam(LEVEL, level);
+        return getResultFromUrl(builder.build().encode().toUri());
+    }
+
+    /**
+     * Fetches the result from a given URL and maps it into a list of LocationDto
+     * objects.
+     *
+     * @param url the URL to fetch results from.
+     * @return a List of LocationDto objects.
+     * @throws RuntimeException if the restTemplate fails to get a response from the
+     *                          provided URL.
+     */
+    public List<LocationDto> getResultFromUrl(URI url) {
+        try {
+            ResponseEntity<Map> response = restTemplate.getForEntity(url, Map.class);
+            return Optional.ofNullable(response)
+                .map(ResponseEntity::getBody)
+                .map(body -> (List<Map<String, Object>>) body.get("results"))
+                .orElseThrow(() -> new NotFoundException(ErrorMessage.NOT_FOUND_LOCATION_BY_URL + url))
+                .stream()
+                .map(this::mapToLocationDto)
+                .collect(Collectors.toList());
+        } catch (RestClientException e) {
+            throw new NotFoundException(ErrorMessage.NOT_FOUND_LOCATION_BY_URL + url);
+        }
+    }
+
+    /**
+     * Maps a result map into a LocationDto object.
+     *
+     * @param result a map containing the result data.
+     * @return a LocationDto object built from the result map.
+     */
+    private LocationDto mapToLocationDto(Map<String, Object> result) {
+        Map<String, String> nameMap = new HashMap<>();
+        nameMap.put("name", getValueFromMap(result, "name"));
```

Comment:
please use the constants you declared here https://github.com/ita-social-projects/GreenCityUBS/pull/1161/files#diff-dc3cf5a2a46824d3495f3c4da780b50ad4ddeebccfc29d59dd1c63d5d89a8152R21